intellij/src/saros/intellij/negotiation/ModuleConfigurationProvider.java

```diff
@@ -0,0 +1,175 @@
+package saros.intellij.negotiation;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.projectRoots.Sdk;
+import com.intellij.openapi.roots.ContentEntry;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.openapi.roots.SourceFolder;
+import com.intellij.openapi.vfs.VirtualFile;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.jps.model.java.JavaResourceRootType;
+import org.jetbrains.jps.model.java.JavaSourceRootType;
+import saros.filesystem.IProject;
+import saros.intellij.filesystem.IntelliJProjectImpl;
+import saros.negotiation.AdditionalProjectDataFactory;
+import saros.negotiation.AdditionalProjectDataFactory.ProjectDataProvider;
+import saros.negotiation.ProjectNegotiationData;
+
+/**
+ * Class providing additional module data needed to create a local representation of the shared
+ * module.
+ */
+public class ModuleConfigurationProvider implements ProjectDataProvider {
+
+  public static final String MODULE_TYPE_KEY = "MODULE_TYPE";
+  public static final String SDK_KEY = "SKD";
+  public static final String SOURCE_ROOTS_KEY = "SOURCE_ROOTS";
+  public static final String TEST_SOURCE_ROOTS_KEY = "TEST_SOURCE_ROOTS";
+  public static final String RESOURCE_ROOTS_KEY = "RESOURCE_ROOTS";
+  public static final String TEST_RESOURCE_ROOTS_KEY = "TEST_RESOURCE_ROOTS";
+
+  private static final String DELIMITER = File.pathSeparator;
```

Comment:
Very bad idea.

Have you read the Javadoc ?

The **system-dependent path-separator** character, represented as a string for convenience. This string contains a single character, namely pathSeparatorChar.

Should be ";" on Windows and ":" on Unix IIRC.

Reply:
Good luck finding a character that is disallowed in filenames under Unix 😆

Reply:
How about `:;` as the delimiter? As at least one of the two is forbidden in either windows or unix, it should be impossible to create a group of paths (in a single system) that breaks the delimiter scheme.

Reply:
Well you could escape like I did with the patch that introduced project name mapping or you dig through some Unix guidelines on how not to name files and compare it with Windows forbidden characters and then pick one that is forbidden under Windows and not recommended to use in filenames under Unix.